### ISSUE TITLE
Upgrade to Swift 1.2

### ISSUE THREAD

**Issue by jspahrsummers** [Carthage/Carthage#326]
This can be done in any release up to and including Carthage 1.0. Some future Swift changes will be unavoidable, but we should leave the gate on the latest version.

**Comment by jspahrsummers** [comment]
Currently blocked on our dependencies—namely, https://github.com/ReactiveCocoa/ReactiveCocoa/issues/1753.

**Comment by jspahrsummers** [comment]
This has to be prioritized for 0.7, because [Madness](https://github.com/robrix/Madness) improvements that we need (namely, [error reporting](https://github.com/robrix/Madness/issues/47)) will only be [available for Swift 1.2](https://github.com/robrix/Madness/pull/58).

/cc @robrix

**Comment by robrix** [comment]
If all else fails and RAC ends up being a blocker for this, we can backport Madness changes to a compatibility branch.

**Comment by jspahrsummers** [comment]
RAC is indeed a blocker for this now, since we weren't able to upgrade to Swift 1.2 yet. :wori

**Comment by robrix** [comment]
Trivial.

**Comment by robb** [comment]
Since Swift 1.2 is now the new hotness, this should maybe get higher priority?

**Comment by danieleggert** [comment]
Swift 1.2 will pretty soon be pretty old with WWDC around the corner.

**Comment by jspahrsummers** [comment]
I invite anyone and everyone interested in :ship:ing this to help resolve the crashes that have plagued #368.

**Comment by danieleggert** [comment]
I hope this doesn’t come off the wrong way (since I sincerely respect all the great work you've done), but I thing having RAC as a dependency to Carthage is a bad decision, and moving off it would make issues like this way easier to resolve. But that's getting a bit OT, and this GitHub issue may not be the best place for that discussion … Carthage is not my project, and hence it's your call if that discussion is worth taking at all. I have no hard feelings about this either way.

**Comment by jspahrsummers** [comment]
ReactiveCocoa is hugely beneficial to the implementation of Carthage, and Carthage was partly conceived as a real world testbed for the RAC 3 API.

Furthermore, RAC isn't really the blocker here. There have been Swift issues since updating to 1.2, like crashes and runtime corruption. Fixing or working around those is what's taking the time.

**Comment by mdiep** [comment]
This is already done. :boom:
